protected/controllers/AdminFileController.php

```diff
@@ -483,41 +483,35 @@ public function actions()
     /**
      * Deletes a particular model.
      * If deletion is successful, the browser will be redirected to the 'admin' page.
+     *
      * @param integer $id the ID of the model to be deleted
      */
-   public function actionDelete($id)
+   public function actionDelete(int $id)
     {
+        // we only allow deletion via POST request
+        if (!Yii::app()->request->isPostRequest) {
+            throw new CHttpException(400, 'Invalid request. Please do not repeat this request again.');
+        }
 
-        if (Yii::app()->request->isPostRequest) {
-            // we only allow deletion via POST request
-            $file = File::model()->findByPk($id);
-            // $file->fileSamples->delete();
-          foreach ($file->fileAttributes as $fileattributes) {
-              print_r($fileattributes);
-              $fileattributes->delete();
-
-          }
-         foreach ($file->fileSamples as $filesample) {
-              print_r($filesample);
-              $filesample->delete();
-
-          }
+        $file = File::model()->findByPk($id);
+        // $file->fileSamples->delete();
 
-         $file->delete();
+        foreach ($file->fileAttributes as $fileattributes) {
+          $fileattributes->delete();
+        }
 
+        foreach ($file->fileSamples as $filesample) {
+            $filesample->delete();
         }
 
-            // if AJAX request (triggered by deletion via admin grid view), we should not redirect the browser
+        $file->delete();
 
-        if (!isset($_GET['ajax'])) {
+        // if AJAX request (triggered by deletion via admin grid view), we should not redirect the browser
+        if (!Yii::app()->request->isAjaxRequest) {
             $this->redirect(isset($_POST['returnUrl']) ? $_POST['returnUrl'] : array('admin'));
```

Comment:
I reckon this part and this variable are not relevant since they are not being received.

Reply:
Hi @alli83 , 
yes, make sense to remove that block if that code is never exercised

Reply:
I think it also needs to handle the case if delete return false or throw an exception. should I open a new ticket?

Reply:
Hi @alli83, you can handle the delete error in this PR. It's a small change, and it's possible for delete to throw exception and return false.